packages/apps/reputation-oracle/server/src/modules/payout/payout.service.ts

```diff
@@ -0,0 +1,227 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { ChainId, EscrowClient } from '@human-protocol/sdk';
+import { PayoutsResultDto } from './payout.dto';
+import { ErrorManifest, ErrorResults } from '../../common/constants/errors';
+
+import {
+  CVAT_JOB_TYPES,
+  CVAT_RESULTS_ANNOTATIONS_FILENAME,
+  CVAT_VALIDATION_META_FILENAME,
+} from '../../common/constants';
+import { ethers } from 'ethers';
+import { Web3Service } from '../web3/web3.service';
+import { JobRequestType } from '../../common/enums';
+import { StorageService } from '../storage/storage.service';
+import { ConfigService } from '@nestjs/config';
+import { CvatManifestDto, FortuneManifestDto } from '../../common/dto/manifest';
+import {
+  CvatAnnotationMeta,
+  FortuneFinalResult,
+  ProcessingResultDto,
+} from '../../common/dto/result';
+
+@Injectable()
+export class PayoutService {
+  private readonly logger = new Logger(PayoutService.name);
+  constructor(
+    @Inject(StorageService)
+    private readonly storageService: StorageService,
+    private readonly web3Service: Web3Service,
+    public readonly configService: ConfigService,
+  ) {}
+
+  /**
+   * Performs payouts for workers based on the calculated results.
+   * It retrieves the signer, builds an escrow client, calculates results,
+   * and initiates bulk payouts through the escrow client.
+   * @param chainId The ID of the blockchain chain.
+   * @param escrowAddress The address of the escrow contract.
+   * @returns {Promise<PayoutsResultDto>} An object containing the URL for results and a boolean indicating if checks passed.
+   */
+  public async executePayouts(
+    chainId: ChainId,
+    escrowAddress: string,
+  ): Promise<PayoutsResultDto> {
+    this.logger.log('Payouts START');
+
+    const signer = this.web3Service.getSigner(chainId);
+    const escrowClient = await EscrowClient.build(signer);
+
+    const results = await this.calculateResults(chainId, escrowAddress);
+
+    await escrowClient.bulkPayOut(
+      escrowAddress,
+      results.recipients,
+      results.amounts,
+      results.url,
+      results.hash,
+      {
+        gasPrice: await this.web3Service.calculateGasPrice(chainId),
+      },
+    );
+
+    this.logger.log('Payouts STOP');
+
+    return {
+      url: results.url,
+      checkPassed: results.checkPassed,
+    };
+  }
+
+  /**
+   * Calculates processing results based on the manifest type.
+   * It retrieves the signer, builds an escrow client, retrieves the manifest URL,
+   * downloads the manifest data, and processes it accordingly.
+   * @param chainId The ID of the blockchain chain.
+   * @param escrowAddress The address of the escrow contract.
+   * @returns {Promise<ProcessingResultDto>} An object containing processing results including recipients, amounts, and storage data.
+   */
+  public async calculateResults(
```

Comment:
Here we can do something like [this](https://github.com/humanprotocol/human-protocol/pull/1574/files)